allow disabling processing for formatters and index names

richm opened the pull request:
The formatters and elasticsearch_index_name configurations now have
an `enabled` parameter.  This defaults to `true`.
This is useful for when you want to have a single plugin
configuration shared among multiple fluentd used as both collectors
and normalizers, and you don't want the collectors to do the
elasticsearch index name creation because they aren't talking directly
to elasticsearch.

richm committed:
allow disabling processing for formatters and index names

The formatters and elasticsearch_index_name configurations now have
an `enabled` parameter.  This defaults to `true`.
This is useful for when you want to have a single plugin
configuration shared among multiple fluentd used as both collectors
and normalizers, and you don't want the collectors to do the
elasticsearch index name creation because they aren't talking directly
to elasticsearch.

richm commented:
@nhosoi @jcantrill @portante PTAL

richm committed:
bump to version 0.0.7

nhosoi commented:
LGTM.

So, in my testing I could add "enabled false" to each <elasticsearch_index_name> in the collector's filter-viaq-data-model.conf?

richm commented:
@nhosoi yes, if you are using the new plugin

richm committed:
preserve @timestamp; improve time handling

There were cases where the existing `@timestamp` was not being
preserved, so make sure we preserve it.

The internal fluentd `time` has only second granularity.  Only use
it where necessary, and use `Time.now` with microsecond granularity
elsewhere.

richm commented:
@nhosoi @jcantrill another commit to review . . .

nhosoi reviewed (approved):
Looks good.
